components/tidb_query/src/codec/convert.rs

```diff
@@ -759,8 +759,10 @@ impl ConvertTo<f64> for &[u8] {
             }
             // if reaches here, it means our code has bug
             Err(err) => {
-                debug_assert!(false);
-                Err(box_err!("parse float err: {}, this is a bug", err))
+                debug_assert!(false, "parse str[{}] to float failed, the err is {}, origin str is: {:?}, this is a bug.", vs, err, self);
+                let err: Error = box_err!("parse str[{}] to float failed, the err is {}, origin str is: {:?}, this is a bug.", vs, err, self);
+                debug!("{:?}", err);
```

Comment:
Prefer to remove L762-L764, the error message has contained entire information, so we don't need the part.

Reply:
I think the `debug_assert!` is necessary, because this `Err(box_err!("Parse '{}' to err: {:?}", vs, err))` can not make the mainater know that this is a bug. And a panic in fuzz test will expose the bug, but a err msg can't.

Reply:
It's the duty of test tools to report the error instead of panic in the program. According to your logic, you need to panic anywhere you return errors.

Reply:
No. If reaches here, then our code has bug. In test, panicing when there is bug is reasonable.

Reply:
TiKV has no unified mechanism to handle bug, so test tool can not find all err that report because of bug.